sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -251,23 +528,21 @@ def plot_confusion_matrix(estimator, X, y_true, *, labels=None,
     >>> plot_confusion_matrix(clf, X_test, y_test)  # doctest: +SKIP
     >>> plt.show()  # doctest: +SKIP
     """
-    check_matplotlib_support("plot_confusion_matrix")
-
+    # early failing to keep the right error message during the deprecation:
```

Comment:
I would suggest to not even change this. 

If we're unlucky and there's a bug in `from_estimator` we would have to backport the bug to this deprecated function

Reply:
So you are suggesting that `plot_confusion_matrix` should fail with a message mentioning `ConfusionMatrixDisplay.from_estimator`?

Reply:
I'm suggesting to not touch `plot_confusion_matrix` at all, apart from the already-added deprecation decorator of course

Reply:
Oh I see then. Yes good point.